<launch>

  <node
      name="ros_traj_consum_interpol_interface"
      pkg="ros_pybullet_interface"
      type="ros_traj_consum_interpol_interface_node.py"
      output="screen">

    <!-- Traj config (required) -->
    <param
        name="traj_config"
        value="{ros_pybullet_interface}/configs/traj_object3D_freemotion.yaml"
        type="str"/>

    <!-- <param
      name="robot_name"
      value="object"
      type="str"/> -->

  </node>

  <node
      name="ros_traj_consum_interpol_interface_sphere1"
      pkg="ros_pybullet_interface"
      type="ros_traj_consum_interpol_interface_node.py"
      output="screen">

    <!-- Traj config (required) -->
    <param
        name="traj_config"
        value="{ros_pybullet_interface}/configs/traj_sphere1.yaml"
        type="str"/>

    <!-- <param
      name="robot_name"
      value="sphere"
      type="str"/> -->

      <!-- <remap from="yang_visual/ros_pybullet_interface/end_effector/traj"  to="yang/ros_pybullet_interface/end_effector/traj" /> -->

  </node>

  <node
      name="ros_traj_consum_interpol_interface_sphere2"
      pkg="ros_pybullet_interface"
      type="ros_traj_consum_interpol_interface_node.py"
      output="screen">

    <!-- Traj config (required) -->
    <param
        name="traj_config"
        value="{ros_pybullet_interface}/configs/traj_sphere2.yaml"
        type="str"/>

    <!-- <param
      name="robot_name"
      value="sphere2"
      type="str"/> -->

     <!-- <remap from="yin_visual/ros_pybullet_interface/end_effector/traj"  to="yin/ros_pybullet_interface/end_effector/traj" /> -->


  </node>

  <node
      name="ros_traj_consum_interpol_interface_Yang"
      pkg="ros_pybullet_interface"
      type="ros_traj_consum_interpol_interface_node.py"
      output="screen">

    <!-- Traj config (required) -->
    <param
        name="traj_config"
        value="{ros_pybullet_interface}/configs/traj_yang.yaml"
        type="str"/>

    <param
      name="robot_name"
      value="yang_visual"
      type="str"/>

    <param
      name="variable_stiffness"
      value="true"
      type="bool"/>

  </node>

  <node
      name="ros_traj_consum_interpol_interface_Yin"
      pkg="ros_pybullet_interface"
      type="ros_traj_consum_interpol_interface_node.py"
      output="screen">

    <!-- Traj config (required) -->
    <param
        name="traj_config"
        value="{ros_pybullet_interface}/configs/traj_yin.yaml"
        type="str"/>

    <param
      name="robot_name"
      value="yin_visual"
      type="str"/>

    <param
      name="variable_stiffness"
      value="true"
      type="bool"/>

  </node>


  <!-- Testing node trajectory publisher -->
  <node name="test_dual_arm_capturing_obj3D"
      pkg="ros_pybullet_interface"
      type="test_dual_arm_catching_FlyingObject.py"
      output="screen">

      <!-- type="test_dual_arm_capturing_prediction.py" -->

<!--      <param-->
<!--          name="rotation_repr"-->
<!--          value="Quat"-->
<!--          type="str"/>-->

      <param
         name="robot1_name"
         value="yang_visual"
         type="str"/>

     <param
        name="robot2_name"
        value="yin_visual"
        type="str"/>

  </node>


</launch>
